File: 1046-last-stone-weight/1046-last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:        
        while len(stones) > 1:
            heavy1 = max(stones)
            stones.remove(heavy1)
            heavy2 = max(stones)
            stones.remove(heavy2)
            
            if heavy1 != heavy2:
                stones.append(abs(heavy1 - heavy2))
        
        if len(stones) == 0:
            return 0
        
        return stones[0]
            
            

File: 1046-last-stone-weight/1046-last-stone-weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int> q;

        for (int i = 0; i < stones.size(); i++) {
            q.push(stones[i]);
        }

        while (q.size() > 1) {
            int first = q.top();
            q.pop();
            int second = q.top();
            q.pop();

            if (first != second) {
                int sum = first - second;
                q.push(abs(sum));
            }
        }
        
        if (q.size() == 0) {
            return 0;
        }
        
        return q.top();
    }
};